### PR TITLE
Feature/set course controller with highest prio as default

### PR DESCRIPTION
**Pull request by tomwim**
### Description
When setting up the training scene the Course Controller was always set to the one in the prefab (Default) instead of the one with the highest priority which is actually intended to be the default loaded one.

**Fixes**: TRNG-1455

### CODE REVIEW

File: Runtime/CourseController/CourseControllerSetup.cs [hunk 1]
```diff
@@ -138,5 +138,15 @@ public static void SetEnforcedCourseController(ICourseController courseControlle
...
+
+        /// <summary>
+        /// Resets the <cref name="courseControllerQualifiedName"/> to the name of the course controller with the highest priority.
+        /// </summary>
+        /// <remarks>This may be used when instantiating a course controller prefab to make sure the default course controller is used.</remarks>
+        public void ResetToDefault()
+        {
+            Type courseControllerType = RetrieveDefaultControllerType();
+            courseControllerQualifiedName = courseControllerType.Name;
+        }
```
Comment (Gusinuhe) [inline Runtime/CourseController/CourseControllerSetup.cs:150]: What about doing this internally in the [Reset](https://docs.unity3d.com/ScriptReference/MonoBehaviour.Reset.html) method?

**Review by Gusinuhe — COMMENTED** [review]
What if th

File: Runtime/CourseController/CourseControllerSetup.cs (hunk 1 above)
Comment (tomwim) [reply Runtime/CourseController/CourseControllerSetup.cs:150]: You cannot call "Reset", I thought about that too :/ Or do you have a way to make this happen?

Comment (tomwim) [reply Runtime/CourseController/CourseControllerSetup.cs:150]: Could remove and add the script again but that also isn't really nice

Comment (Gusinuhe) [reply Runtime/CourseController/CourseControllerSetup.cs:150]: True 🤔
Maybe calling it with reflection? Or making it internal?

Consider this as a suggestion, nothing that blocks merging this, I just wonder a way not to expose unnecessary APIs.

Comment (tomwim) [reply Runtime/CourseController/CourseControllerSetup.cs:150]: I tried it with reflection and it technically works but makes things very complicated with the editor script and probably will lead to some bigger changes. so i would keep it like this

**Review by Gusinuhe — APPROVED** [review]

**Pull request merged**